src/ert/storage/migration/to9.py

```diff
@@ -0,0 +1,87 @@
+import json
+import os
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+import polars
+
+info = "Migrate finalized response keys into configs"
+
+
+def _write_transaction(filename: str | os.PathLike[str], data: bytes) -> None:
+    """
+    Writes the data to the filename as a transaction.
+
+    Guarantees to not leave half-written or empty files on disk if the write
+    fails or the process is killed.
+    """
+
+    Path("./swp").mkdir(parents=True, exist_ok=True)
+    with NamedTemporaryFile(dir="./swp", delete=False) as f:
+        f.write(data)
+        os.rename(f.name, filename)
+
+
+def migrate(path: Path) -> None:
+    for experiment in path.glob("experiments/*"):
+        ensembles = path.glob("ensembles/*")
+
+        with (
+            open(experiment / "index.json", encoding="utf-8") as f_experiment,
+            open(
+                experiment / "responses.json", mode="r", encoding="utf-8"
+            ) as f_responses,
+        ):
+            exp_index = json.load(f_experiment)
+            experiment_id = exp_index["id"]
+
+            responses_config = json.load(f_responses)
+            for response_type, config in responses_config.items():
+                if "has_finalized_keys" not in config:
+                    # Read a sample response and write the keys
+                    for ens in ensembles:
```

Comment:
Perhaps just pick the first one? Think that would simplify the logic somewhat?

Reply:
I'm not completely sure, wrt edge cases / failed ensembles, unless we can somehow pick the "prior" ensemble always (without the complexity of finding the ensemble with iter=0 which is similar to the complexity of break'ing out)